.github/workflows/integration-test.yml

```diff
@@ -105,3 +104,41 @@ jobs:
         # For every other version and/or OS, run pytest without coverage
         if: (matrix.os != 'ubuntu-latest') || (matrix.python-version != env.TARGET_PYTHON_VERSION )
         run: poetry run pytest ${{ github.event.inputs.pytest_addopts }}
+  build:
+    name: Build versioned package
+    # This job is used to build the package and upload the artifacts
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.get_version.outputs.version }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Setup python, and load cache
+        uses: ./.github/actions/setup-env
+        with:
+          python-version: ${{ env.TARGET_PYTHON_VERSION }}
+          cache-pre-commit: false
+          cache-venv: true
+          setup-poetry: true
+          install-deps: true
+      - name: Build
+        run: poetry build
+      - name: Get pycytominer version
+        id: get_version
+        run: |
+          echo "version=$(poetry version | cut -d' ' -f2 )"
+          echo "version=$(poetry version | cut -d' ' -f2 )" >> "$GITHUB_OUTPUT"
+      - name: Upload artifacts
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+        with:
```

Comment:
Consider setting an explicit retention period via `retention-days` (here and below) to help indicate how long the artifacts might be available. Depending on the usage intention, I wonder if non-default, shorter periods of time might be helpful to avoid exceeding the storage quota. Open to other thoughts here too.

Reply:
Best I can tell from the Github documentation, there is no artifact storage limit for public repos. The storage limits in the table [here](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions) for Github Free accounts only apply to private repos. 

I suggest we hold off on adding a retention policy until if/when we encounter issues.

Reply:
Mostly here I suggest being as specific as we can when it comes to artifact retention and storage considerations. This is an element we can help reduce the [uncertainty of GitHub's mysteriousness in this area](https://github.com/actions/upload-artifact/issues/9) through explicit definition. The [default retention policy is 90 days](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#artifact-and-log-retention-policy), and informing Pycytominer developers upfront I feel would be helpful if it became an issue of total storage considerations or planning for other tooling.